### ISSUE TITLE
Article content is truncated

### ISSUE THREAD

**Issue by kelson42** [automactic/Kiwix#30]
If you look at the footer, you see the sentence of the footer is often truncated. I have tested with wikipedia_lij_all_nopic_2015-04.zim. This file loaded with kiwix-serve does not suffer of the same problem, so it looks like to be a problem by the loading.

**Comment by automactic** [comment]
I know why this is happening. The contentLength returned by getContentByUrl is shorter than what it actually is. 

Here is the code:
- (NSData *)dataWithContentURLString:(NSString *)contentURLString {
    NSData *contentData;
    
    string pageURLStringC = [contentURLString cStringUsingEncoding:NSUTF8StringEncoding];
    string content;
    string contentType;
    unsigned int contentLength = 0;
    if (_reader->getContentByUrl(pageURLStringC, content, contentLength, contentType)) {
        contentData = [NSData dataWithBytes:content.data() length:contentLength];
    }
    NSLog(@"zimReader: URL passed to getContentByUrl(): %@, getDataLength: %lu",contentURLString, (unsigned long)[contentData length]);
    return contentData;
}

**Comment by automactic** [comment]
I tried to fix this using contentData = [NSData dataWithBytes:content.data() length:floor(contentLength*1.024)]; Since apple use 1000Bytes = 1KB instead 1024. But no good, sometimes it will crash due to accessing something not there. And sometimes it will cause gibberish appear in the rear of the whole untruncated article.

**Comment by kelson42** [comment]
I would really wonder if getContentByUrl() return the wrong length in bytes. Please give me a concrete example if you find one.

**Comment by automactic** [comment]
OK, this one is fixed.

I replaced contentLength returned by getContentByUrl() with strlen(content.c_str())

**Comment by kelson42** [comment]
This is definitly the wrong way because:
1 - you have no garanty that the string does not have a zero in the middle, and then strlen will fail
2 - the value returned by getContentByUrl() is correct, if not we have a bug and this needs ot be clearly identied.

I'm pretty sure this kind of fix does not work with pictures.

**Comment by automactic** [comment]
Yes, you are right. They don't work with pictures. I have changed it back.
